src/components/solvers/ProgressHandler.tsx

```diff
@@ -1,33 +1,70 @@
-import { Box, Center, HStack, VStack } from "@chakra-ui/react";
-import React, { useState } from "react";
+import { Box, Center, HStack } from "@chakra-ui/react";
+import React, { useEffect, useState } from "react";
+import { History } from "./History";
 import { GoButton } from "./buttons/GoButton";
-import { postProblem } from "../../api/ToolboxAPI";
+import { fetchSolution, postProblem } from "../../api/ToolboxAPI";
 import { SolutionView } from "./SolutionView";
 import { Container } from "../Container";
 import { Solution } from "../../api/data-model/Solution";
 import { SolveRequest } from "../../api/data-model/SolveRequest";
 import { SolverPicker } from "./SolverPicker";
 
+export interface ProblemState<T> {
+  /**
+   * Problem input
+   */
+  content: T;
+  /**
+   * Ids of the solutions of the problem input per problem type
+   */
+  solutionIds: ProblemTypeSolutionId;
+}
+
+export type ProblemTypeSolutionId = {
+  [problemTypeId: string]: number;
+};
```

Comment:
The name is a bit misleading. It seems like an object of this type is an ID. Please give this a clearer name (i.e., `SolutionIds`) or just use a `Map`